### PR TITLE
Feature/access ingredient and pantry information

### PR DESCRIPTION
**Pull request by lschnell8**
PR Template:

Is this a fix or a feature?
- [ ] Bug fix
- [x] Feature
- [ ] Refactor
 
Describe the feature - fix - refactor:
Add methods to Pantry class and make adjustments to integrate.

How was this tested?
- [ ] open index.html
- [ ] localhost
- [x] terminal

Where should the reviewer start - test:
Pantry.js

### CODE REVIEW

File: src/classes/Pantry.js [hunk 1]
```diff
@@ -1,46 +1,72 @@
...
+      if (!enoughIngredients) {
+        return ingredient
+        //returns an array of recipe ingredient objects that are missing from the pantry or less than the amount needed to make the recipe
+      }
+    })
+    return ingredientsNeeded
+  }
+
+  findAmountNeeded(recipe) {
+    let recipeIngredients = recipe.findRecipeIngredientInfo();
```
Comment (StevieAmb) [inline src/classes/Pantry.js:33]: Teamwork makes the dream work!

File: src/classes/Pantry.js [hunk 2]
```diff
@@ -1,46 +1,72 @@
...
+          recipeIngredient.amountNeeded = pantryIngredient.amount + (parseInt(recipeIngredient.quantity.amount) - pantryIngredient.amount)
+        }
+        if (recipeIngredient.id !== pantryIngredient.ingredient) {
+          recipeIngredient.amountNeeded = recipeIngredient.quantity.amount
         }
       })
+        return recipeIngredient
     })
-    return userMessage
+    return newShelf.filter(ingredient => ingredient.amountNeeded)
```
Comment (StevieAmb) [inline src/classes/Pantry.js:45]: This was cool to figure out how to do, and figure out how to get the information we wanted out of it.

File: src/scripts.js [hunk 3]
```diff
@@ -260,7 +260,7 @@ const showRecipeInformation = (event) => {
       ingredientsList.innerHTML = ``  
       recipeIngredients.forEach((ingredient) => {
         ingredientsList.insertAdjacentHTML('beforeEnd', `
-        <li>${ingredient} </li>`)
+        <li>${ingredient.quantity.amount} ${ingredient.quantity.unit} ${ingredient.name}</li>`)
```
Comment (StevieAmb) [inline src/scripts.js:265]: Yes, display!!

**Review by StevieAmb — APPROVED** [review]

File: src/classes/Pantry.js (hunk 1 above)
Comment (Alynn022) [reply src/classes/Pantry.js:33]: This looks clean!

File: test/Pantry-test.js [hunk 4]
```diff
@@ -19,24 +19,210 @@ describe.only('Recipe', () => {
...
+            },
+            {
+              id: 2050,
+              quantity: {
+                amount: 0.5,
+                unit: 'tsp'
+              },
+              name: 'vanilla'
+            }
+          ])
```
Comment (Alynn022) [inline test/Pantry-test.js:115]: I like how detailed this test is

**Review by Alynn022 — APPROVED** [review]

**Pull request merged**